extracted-files/grammar.abnf

```diff
@@ -59,11 +59,11 @@ Enum    = Tag
 
 ; ------------- Date -------------
 
-DateValue   = date / DatePeriod / dateRange / dateApprox / ""
+DateValue   = [ date ] / DatePeriod / dateRange / dateApprox
 DateExact   = day D month D year  ; in Gregorian calendar
-DatePeriod  = %s"FROM" D date [D %s"TO" D date]
-            / %s"TO" D date
-            / ""
+DatePeriod  = [ %s"TO" D date ]
+            / %s"FROM" D date [D %s"TO" D date]
+            ; note both DateValue and DatePeriod can be the empty string
```

Comment:
would this be the same as the simpler form
```suggestion
DatePeriod  = [%s"FROM" D date] [D %s"TO" D date]
```
?

Reply:
That's simpler, but not the same; the former encodes "`TO 2022`" but the latter " TO 2022`" (note the extra space)